env/work/srw_python/uti_plot_matplotlib.py

```diff
@@ -127,12 +161,27 @@ def uti_data_file_plot(self, _fname, _read_labels=1, _e=0, _x=0, _y=0, _graphs_j
         #  fig = __mode_EH(data,allrange)
         elif mode==m.EHV:
             fig = self.__mode_EHV(data, allrange, arLabels, arUnits, _e, _x, _y, _graphs_joined)
+        if _multicolumn_data: #MR31102017
+            available_cols = list(data.keys())
+            for c in [_column_x, _column_y]:
+                if c not in available_cols:
+                    raise ValueError('Incorrect column specified: {}.\nAvailable columns: {}'.format(c, available_cols))
+            fig = self._plot_1D_XvsY(data, _column_x, _column_y)
         return self._maybe_savefig(fig)
 
     def _enum(self, *sequential, **named): #Had to copy this in uti_plot_com
         enums = dict(zip(sequential, range(len(sequential))), **named)
         return type('Enum', (), enums)
 
+    def _plot_1D_XvsY(self, data, column_x, column_y, figsize=(8, 5), typ=111): #MR31102017
```

Comment:
Actually, this function would rather be called "_plot_1D_YvsX". But OK, I left it "as is" for the moment.